onnx/optimizer/passes/fuse_pad_into_conv.h

```diff
@@ -0,0 +1,99 @@
+// ATTENTION: The code in this file is highly EXPERIMENTAL.
+// Adventurous users should note that the APIs will probably change.
+
+#pragma once
+
+// Before:
+//   P = Pad(X)
+//   Z = Conv(P, Y)
+// After:
+//   Z = Conv(X, Y) with "pads" attribute set
+//
+// the pass handles the case when Pad is zero-padding the input
+// (i.e. mode=constant and value=0)
+
+#include <numeric>
+
+#include "onnx/common/assertions.h"
+#include "onnx/optimizer/pass.h"
+
+namespace ONNX_NAMESPACE {
+namespace optimization {
+
+struct FusePadIntoConv final : public PredicateBasedPass {
+  explicit FusePadIntoConv()
+      : PredicateBasedPass(
+            PassType::Fuse,
+            PassEfficiency::Complete,
+            PassOptimizationType::Compute) {}
+  std::string getPassName() const override {
+    return "fuse_pad_into_conv";
+  }
+  bool patternMatchPredicate(Node* node) override {
+    return node->kind() == kConv && node->inputs()[0]->node()->kind() == kPad;
+  }
+  bool runTransform(Node* n, Graph& graph, NodeDestroyType& destroy_current)
```

Comment:
It seems the "graph" here has not been used, and that's why the CI failed.
Wonder if it is designed to be like this?

Reply:
I noticed and fixed in the latest commit, but I don't understand how is this not an issue in other optimizers, like `fuse_consecutive_squeezes` or `eliminate_nop_pad`?

Reply:
@zrphercule I believe that CI checks for unused parameters.

@vloncar both fuse_consecutive_squeezes and eliminate_nop_pad have definitions which don't use graph:

`bool runTransform(Node* n, Graph&, NodeDestroyType& destroy_current)`